src/components/cards/ArnsCard/ArnsCard.tsx

```diff
@@ -0,0 +1,70 @@
+import './styles.css';
+import { ArnsCardProps } from '../../../types';
+import { useEffect, useState } from 'react';
+import { useStateValue } from '../../../state/state';
+import { ArnsDefault } from '../../icons';
+import axios from 'axios';
+
+function ArnsCard({ name }: ArnsCardProps) {
+  const [{ arnsSourceContract }] = useStateValue();
+  const [antDetails, setAntDetails] = useState({
+    name: 'arns',
+    gateway: 'arweave.net',
+    antContractAddress: '',
+    image: '',
+    expiry: '',
+  });
+
+  useEffect(() => {
+    getAntDetailsFromName(name);
+  }, []);
+
+  async function getAntDetailsFromName(name: string) {
+    const antAddress = arnsSourceContract.records[name];
+    const expiry = new Date().toDateString();
+    const imageUrl = await getMetaImage();
+
+    setAntDetails({
+      name: name,
+      gateway: 'arweave.dev',
+      antContractAddress: antAddress,
+      image: imageUrl,
+      expiry: expiry,
+    });
+  }
+  async function getMetaImage() {
+    const img = await axios
+      .get(`http://${name}.arweave.dev`)
+      .then((res) => res.data)
+      .then((html) => {
+        let parser = new DOMParser();
+        let doc = parser.parseFromString(html, 'text/html');
+        let metaImage = doc
+          ?.querySelector("meta[property='og:image']")
+          ?.getAttribute('content');
+
+        return metaImage;
+      });
+    const status = await axios
+      .get(img!)
+      .then((res) => res.status)
+      .catch((error) => console.log(error));
+    if (status === 200) {
+      return `${img}`;
+    } else {
+      return `${ArnsDefault}`;
+    }
```

Comment:
we should wrap this whole thing in a a try catch and by default set the image to `arnsDefault` on load while we fetch the og tags